Framework/Kernel/test/ArrayPropertyTest.h

```diff
@@ -301,6 +301,65 @@ class ArrayPropertyTest : public CxxTest::TestSuite {
                       static_cast<Property *>(0))
   }
 
+  void testPrettyPrinting() {
+    std::vector<std::string> inputList{
+        "1,2,3", "-1,0,1", "356,366,367,368,370,371,372,375", "7,6,5,6,7,8,10",
+        "1-9998, 9999, 2000, 20002-29999"};
+    std::vector<std::string> resultList{"1-3", "-1-1",
+                                        "356,366-368,370-372,375", "7,6,5-8,10",
+                                        "1-9999,2000,20002-29999"};
+
+    TSM_ASSERT("Test Failed for vectors of int",
+               listShorteningwithType<int>(inputList, resultList));
+    TSM_ASSERT("Test Failed for vectors of long",
+               listShorteningwithType<long>(inputList, resultList));
+    // explicit test for in32_t with matches det_id_t and spec_id_t
+    TSM_ASSERT("Test Failed for vectors of int32_t",
+               listShorteningwithType<int32_t>(inputList, resultList));
+
+    // unsigned types
+    std::vector<std::string> inputListUnsigned{
+        "1,2,3", "356,366,367,368,370,371,372,375", "7,6,5,6,7,8,10",
+        "1-9998, 9999, 2000, 20002-29999"};
+    std::vector<std::string> resultListUnsigned{
+        "1-3", "356,366-368,370-372,375", "7,6,5-8,10",
+        "1-9999,2000,20002-29999"};
+    TSM_ASSERT("Test Failed for vectors of unsigned int",
+               listShorteningwithType<unsigned int>(inputListUnsigned,
+                                                    resultListUnsigned));
+    TSM_ASSERT(
+        "Test Failed for vectors of size_t",
+        listShorteningwithType<size_t>(inputListUnsigned, resultListUnsigned));
+
+    // check shortening does not happen for floating point types
+    std::vector<std::string> inputListFloat{
+        "1.0,2.0,3.0", "1.0,1.5,2.0,3.0", "-1,0,1",
+    };
+    std::vector<std::string> resultListFloat{
+        "1,2,3", "1,1.5,2,3", "-1,0,1",
+    };
+    TSM_ASSERT("Test Failed for vectors of float",
+               listShorteningwithType<float>(inputListFloat, resultListFloat));
+    TSM_ASSERT("Test Failed for vectors of double",
+               listShorteningwithType<double>(inputListFloat, resultListFloat));
+  }
+
+  template <typename T>
+  bool listShorteningwithType(const std::vector<std::string> &inputList,
+                              const std::vector<std::string> &resultList) {
+
+    bool success = true;
+    for (size_t i = 0; i < inputList.size(); i++) {
+      ArrayProperty<T> listProperty("i", inputList[i]);
+      std::string response = listProperty.valueAsPrettyStr(0, true);
+      TS_ASSERT_EQUALS(response, resultList[i]);
+      if (response != resultList[i]) {
+        success = false;
```

Comment:
Actually here I would rather it ran through all of the test cases, the TS_ASSERT_EQUALS will output useful feedback for each test case that fails.